godot-core/src/tools/autoload.rs

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use crate::builtin::NodePath;
+use crate::classes::{Engine, Node, SceneTree};
+use crate::meta::error::ConvertError;
+use crate::obj::{Gd, Inherits, Singleton};
+
+/// Retrieves an autoload by its name.
+///
+/// See [Godot docs] for an explanation of the autoload concept. Godot sometimes uses the term "autoload" interchangeably with "singleton";
+/// we strictly refer to the former to separate from [`Singleton`][crate::obj::Singleton] objects.
+///
+/// [Godot docs]: https://docs.godotengine.org/en/stable/tutorials/scripting/singletons_autoload.html
+///
+/// # Panics
+/// This is a convenience function that calls [`try_get_named_autoload()`]. Panics if that fails, e.g. not found or wrong type.
+///
+/// # Example
+/// ```no_run
+/// use godot::prelude::*;
+///
+/// #[derive(GodotClass)]
+/// #[class(init, base=Node)]
+/// struct GlobalStats {
+///     base: Base<Node>,
+/// }
+///
+/// // Assuming "Statistics" is registered as an autoload in `project.godot`,
+/// // this returns the one instance of type Gd<GlobalStats>.
+/// let stats = get_named_autoload::<GlobalStats>("Statistics");
+/// ```
+pub fn get_named_autoload<T>(autoload_name: &str) -> Gd<T>
```

Comment:
I would argue that in current form it doesn't bring that much improvement over `get_node_as<T>("root/xyz")` – while being more taxing for Nodes.

Reply:
Yes, I'll add caching in a 2nd commit once the base one works as expected.